.github/workflows/release.yml

```diff
@@ -1,157 +1,232 @@
-          brew install elixir
-          mix local.rebar --force
-          mix local.hex --force
-
-      - name: Retrieve Mix Dependencies Cache
-        if: ${{ !startswith(matrix.os, 'windows') }}
-        uses: actions/cache@v2
-        id: mix-cache #id to use in retrieve action
-        with:
-          path: |
-            wadm/deps
-            wadm/_build
-          key: ${{ runner.os }}-${{ matrix.otp }}-${{ matrix.elixir }}-mix-${{ hashFiles('wadm/mix.exs', 'wadm/mix.lock') }}
+      - uses: actions/checkout@v3
 
-      - name: Install Mix Dependencies
-        working-directory: ${{env.working-directory}}
-        if: steps.mix-cache.outputs.cache-hit != 'true'
-        run: |
-          mix do deps.get, compile
+      - name: set the release version (tag)
+        if: startsWith(github.ref, 'refs/tags/v')
+        shell: bash
+        run: echo "RELEASE_VERSION=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: set the release version (main)
+        if: github.ref == 'refs/heads/main'
+        shell: bash
+        run: echo "RELEASE_VERSION=canary" >> $GITHUB_ENV
 
-      - name: Create Mix Release
-        working-directory: ${{env.working-directory}}
+      # To manually test workflow
+      - name: set the release version (manually trigger test with canary tag push)
+        if: startsWith(github.ref, 'refs/tags/canary')
         shell: bash
-        run: mix release wadm
+        run: echo "RELEASE_VERSION=canary" >> $GITHUB_ENV
 
-      - name: TarGEEZY Mix Release
-        working-directory: ${{env.working-directory}}/_build/prod/rel/wadm
+      - name: lowercase the runner OS name
         shell: bash
         run: |
-          tar -czvf ${{ matrix.release-tarball-filename }} bin erts-* lib releases
-          cp ${{ matrix.release-tarball-filename }} ../../../../
+          OS=$(echo "${{ runner.os }}" | tr '[:upper:]' '[:lower:]')
+          echo "RUNNER_OS=$OS" >> $GITHUB_ENV
 
-      - name: Upload artifact
-        uses: actions/upload-artifact@v2
+      - name: Install latest Rust stable toolchain
+        uses: dtolnay/rust-toolchain@stable
+        if: matrix.config.arch != 'aarch64'
         with:
-          name: ${{ matrix.release-tarball-filename }}
-          path: ${{env.working-directory}}/${{ matrix.release-tarball-filename }}
+          toolchain: stable
+          components: clippy, rustfmt
 
-  release-docker:
-    if: startswith(github.ref, 'refs/tags/') # Only run on tag push
-    name: Release Linux Docker Image
-    runs-on: ubuntu-latest
-    env:
-      MIX_ENV: prod
-      working-directory: wadm
-
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
+      - name: setup for cross-compile builds
+        if: matrix.config.arch == 'aarch64' && matrix.config.os == 'ubuntu-20.04'
+        run: |
+          sudo apt-get update
+          sudo apt install gcc-aarch64-linux-gnu g++-aarch64-linux-gnu
+          rustup toolchain install stable-aarch64-unknown-linux-gnu
```

Comment:
I have tried to condense this step with the following "Install Rust Stable Toolchain Step", however, it does not work. Not really sure why, and don't want to spend much more time on this. I have tried changing step order as well. For now this works, but it seems redundant to have this step. May be nice to condense in future.